subprojects/problems/src/main/java/org/gradle/api/problems/ProblemBuilderDefiningCategory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems;
+
+import org.gradle.api.Incubating;
+
+/**
+ * {@link Problem} instance builder requiring the specification of the problem type.
+ *
+ * @since 8.5
+ */
+@Incubating
+public interface ProblemBuilderDefiningCategory {
+
+    /**
+     * Declares the problem category string.
+     * The category string is a ":" separated string that starts either with a specific problem category or with "gradle-plugin"
+     * and the plugin id where the problem is raised followed by the specific category.
+     * <pre>{@code
+     *  structure:
+     *  <category>:<detailed-info>:<detailed-info>:...
+     *  gradle-plugin:<plugin-id>:<category>:<detailed-info>:<detailed-info>:...
+     * }</pre>
+     *
+     * <p>examples of valid category strings:
+     * <pre>{@code
+     * deprecation
+     * gradle-plugin:deprecation
+     * gradle-plugin:deprecation:<detailed-info>
+     * gradle-plugin:deprecation:<detailed-info>:<detailed-info>:...
+     * }</pre>
+     *
+     * @param category the type name
+     * @param details the type details (these will be appended to the category string and separated by ":")
+     * @return the builder for the next required property
+     */
```

Comment:
❌ We are exposing too many details here. I think we should only document what's necessary for the problem producers. This would be especially confusing for plugin authors as they might think that the plugin id should be manually baked into the reports (the problems service or the future public API variant should take care of it). The structure representation might make sense on the ProblemCategory interfaces or on the future transformers and TAPI interfaces.

Draft for javadoc:
```suggestion
     * It offers a hierarchical categorization with arbitrary details.
     * Clients must declare a main category string. Freeform with the following conventions and limitations.
     * Subcategories can be optionally specified with arbitrary details. The same conventions and limitations apply.
     * When a problem is created (with BuildableProblemBuilder.build()) the category can be obtained with {@link Problem.getCategory()}. The `ProblemCategory` then represents a local category with some namespace, distinguishing built-in and third-party categories.
     * Example: {@code category("validation", "missing-input")
     
     * @param category the type name
     * @param details the type details
     * @return the builder for the next required property
     * @see ProblemCategory
     */
```